weave/flow/monitor.py

```diff
@@ -0,0 +1,119 @@
+from pydantic import Field, field_validator
+from typing_extensions import Self
+
+from weave.flow.casting import ScorerLike
+from weave.flow.obj import Object
+from weave.trace.api import ObjectRef, publish
+from weave.trace.objectify import register_object
+from weave.trace.vals import WeaveObject
+
+
+@register_object
+class Monitor(Object):
+    """
+    Sets up a monitor to score incoming calls automatically.
+
+    Examples:
+
+    ```python
+    import weave
+    from weave.scorers import ValidJSONScorer
+
+    json_scorer = ValidJSONScorer()
+
+    my_monitor = weave.Monitor(
+        name="my-monitor",
+        description="This is a test monitor",
+        sampling_rate=0.5,
+        call_filter={
+            "op_names": ["my_op"],
+            "query": {
+                "$expr": {
+                    "$gt": [
+                        {
+                            "$getField": "started_at"
+                        },
+                        {
+                            "$literal": 1742540400
+                        }
+                    ]
+                }
+            }
+        },
+        scorers=[json_scorer],
+    )
+
+    my_monitor.activate()
+    ```
+    """
+
+    sampling_rate: float = Field(ge=0, le=1)
+    scorers: list[ScorerLike]
+    call_filter: dict
```

Comment:
is it possible to type this as a subset of the `CallsQueryReq`. Perhaps the filter and query fields? I think that might provide better alignment with the query layer. see `weave/trace_server/interface/builtin_object_classes/saved_view.py` for a similar style

Reply:
Instead I added a manual Pydantic validation to make sure the query is properly formatted.

Reply:
The problem you are observing is due to what I view as a bug on the python client read-side. Idk if you want to make a decision on type spec due to a client reading bug. These WeaveObjects are frustrating for a number of reasons. Howe are you getting `query` here?